libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -530,7 +530,7 @@ static void _destroyMqttConnection( _mqttConnection_t * pMqttConnection )
      * This can happen in case of an error in #IotMqtt_Connect() API. */
     if( contextIndex == -1 )
     {
-        IotLogWarn( "(MQTT connection %p) does not have an MQTT Context." );
+        IotLogWarn( "(MQTT connection %p) does not have an MQTT Context.", pMqttConnection );
```

Comment:
Under what normal conditions could this happen?  This seems like something that shouldn't happen.  Should this be an error or even an `assert()`?

Reply:
This can happen in a couple of error paths in IoTMqtt_Connect call.
 1. An MQTTConnection can be allocated but failed to allocate a corresponding MQTT Context.
 2. Setting a receive callback fails in the network interface.

This function will clean up the memory allocated for MQTTConnection even if the MQTTContext was not allocated.